src/ServiceControl.Monitoring/Timings/ProcessingTimeStore.cs

```diff
@@ -1,7 +1,28 @@
 ﻿namespace ServiceControl.Monitoring.Timings
 {
-    public class ProcessingTimeStore : TimingsStore
+    using System;
+    using System.Linq;
+    using Infrastructure;
+    public class ProcessingTimeStore : IntervalsStore, IProvideEndpointMonitoringData, IProvideEndpointInstanceMonitoringData
     {
-        
+        public void FillIn(MonitoredEndpoint[] data, DateTime now)
+        {
+            var snapshot = GetIntervals(now).ToLookup(x => x.Id.EndpointName);
+
+            foreach (var item in data)
+            {
+                item.ProcessingTime = AggregateTimings(snapshot[item.Name].ToList());
+            }
+        }
+
+        public void FillIn(MonitoredEndpointInstance[] data, DateTime now)
+        {
+            var snapshot = GetIntervals(now).ToLookup(k => $"{k.Id.EndpointName}-{k.Id.InstanceId}");
```

Comment:
If the view model included a proper EndpointInstanceId then we could just lookup based on that.